replace_with/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+// Julien Chien <jchien17@cmc.edu>
+
+pub fn replace_with<T, F: FnOnce(T) -> T>(t: &mut T, f: F) {
+  unsafe {
+    let mut ptr: *mut T = t;
+    *ptr = f(std::ptr::read(ptr))
```

Comment:
I get 

```
$ cargo test --test required
    Finished debug [unoptimized + debuginfo] target(s) in 0.0 secs
     Running target/debug/required-1792531e33f78f4c

running 2 tests
test required::complex_type ... ok
test required::simple_type ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured
```

I don't see the `error` after the `test result` section, maybe we have different versions of rust?